lib/Dialect/ATen/ATenDialectOpStats.cpp

```diff
@@ -0,0 +1,826 @@
+//===- ATenDialectOpStats.cpp -----------------------------------*- C++ -*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "npcomp/Dialect/ATen/ATenDialect.h"
+#include "npcomp/Dialect/ATen/ATenOpStatisticsUtils.h"
+
+#include "llvm/Support/Debug.h"
+
+#include "mlir/IR/StandardTypes.h"
+#include "mlir/IR/Types.h"
+
+#include <iostream>
+
+#define DEBUG_TYPE "aten-op-stats"
+
+// This file contains the StatisticsOpInterface implementations
+// for ATDialect operations
+
+using namespace mlir;
+
+namespace {
+
+std::vector<uint64_t> unpackListConstant(Value op) {
+  std::vector<uint64_t> v;
+  auto co = cast<mlir::NPCOMP::aten::ConstantOp>(op.getDefiningOp());
+  DenseElementsAttr a = co.template getAttrOfType<DenseElementsAttr>("value");
+  for (auto i : a.getIntValues())
+    v.push_back(i.getSExtValue());
+  return v;
+};
+
+} // namespace
+
+namespace mlir {
+namespace NPCOMP {
+namespace aten {
+
+std::map<std::string, uint64_t> AdaptiveAvgPool2dOp::getStatistics() {
+  std::map<std::string, uint64_t> toReturn;
+  // FIXME: unimplemented
+  toReturn["reads"] = -1;
```

Comment:
I wish we had made the statistics map be an int64_t (which would make more sense for the -1 here). For a followup.

Reply:
I've added an issue to look at this design from scratch.